### PR TITLE
feat: Add sprite prefab

### LINKED ISSUE #1149: Allow SpriteSheet .ron files to load grid like spritesheets
Not sure if this was already discussed once but either way.

I'd like to propose a different way of how a spritesheet would be loaded from a .ron file. Most engines have this option of a grid like importing, so why not do this with .ron files too. Maybe this is already possible or not but I'm just going to give an example below. 

This would help me atleast in being able to define a spritesheet quicker and without having to define 50+ sprites that only have x and y coordinates changed. The example below has 8 sprites inside and this requires me to calculate the x everytime which takes a bit.

    spritesheet_width: 576.0,
    spritesheet_height: 40.0,
    sprites: [
        (
            x: 0.0,
            y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
		(
            x: 72.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
		(
            x: 144.0,
            y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
		(
            x: 216.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
	(
            x: 288.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
	(
            x: 360.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
	(
            x: 432.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
	(
            x: 504.0,
	    y: 0.0,
            width: 72.0,
            height: 40.0,
	    offsets: Some((-32.0, -16.0))
        ),
    ],

An alternative to this would simply be this:

	spritesheet_width: 576.0,
        spritesheet_height: 40.0,
	columns: 8,
	rows: 1,
	cell_size: Some((72.0, 40.0))
        offsets: Some((-32.0, -16.0))

Sorry for the horrible indentation 😄 

So maybe this could be added as a seperate option for spritesheets with a specifier in the ron file?

### PR DESCRIPTION
**Pull request by Rhuagh**
Fixes #1149.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/1352)

### CODE REVIEW

**Comment by Rhuagh** [comment]
Note that this is not ready yet, need to fix documentation and some example ron files.

File: amethyst_renderer/src/sprite/prefab.rs [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+    fn grid_col_row() {
+        let sprites = SpriteGrid {
+            width: 400,
+            height: 200,
+            columns: 4,
+            rows: Some(4),
+            ..Default::default()
+        }
+        .build_sprite_list();
+        println!("{:?}", sprites);
```
Comment (azriel91) [inline amethyst_renderer/src/sprite/prefab.rs]: probably meant to be deleted

**Comment by vessd** [comment]
@Rhuagh it looks very cool
Do you need help? Maybe I can help with something.

**Comment by azriel91** [comment]
Superseded and done in #1469.

**Pull request closed without merging**